themes/loop/template.php

```diff
@@ -821,10 +824,60 @@ function loop_theme($existing, $type, $theme, $path) {
       'path' => drupal_get_path('theme', 'loop') . '/templates/forms',
       'template' => 'comment-form-prefix',
     ),
+    'user_login' => array(
+      'render element' => 'form',
+      'path' => drupal_get_path('theme', 'loop') . '/templates/user',
+      'template' => 'user-login',
+      'preprocess functions' => array(
+        'loop_preprocess_user_login',
+      ),
+    ),
   );
 }
 
 /**
+ * User login form.
+ */
+function loop_preprocess_user_login(&$variables) {
+  $login_services = array();
+
+  $destination = drupal_get_destination();
+  $returnTo = $destination['destination'];
+  $options = array('query' => array('returnTo' => $returnTo));
+
+  // Saml log in services.
+  if (function_exists('saml_sp__load_all_idps')) {
```

Comment:
I'm not changing or moving anything, but simply building a list of active login services to be rendered in the login template. I've added some comments to make it more clear what is going on.